src/main/java/org/apache/accumulo/testing/performance/tests/GroupCommitPT.java

```diff
@@ -0,0 +1,243 @@
+      // Each thread flushes after each mutation.
+      for (int numThreads : tests) {
+        runBatch(env, report, numThreads, 1, false, walog);
+      }
+
+      // This test a single thread write a different batch sizes of mutations, flushing after each
+      // batch.
+      // Group commit should approach these times for the same number mutations.
+      for (int numToBatch : tests) {
+        runBatch(env, report, 1, numToBatch, false, walog);
+      }
+
+      walog = !walog;
+    }
+
+    return report.build();
+  }
+
+  private void runBatch(Environment env, Report.Builder report, int numThreads, int numToBatch,
+      boolean useSharedBW, boolean walog) throws Exception {
+
+    String tableName = "mutslam";
+
+    try {
+      env.getClient().tableOperations().create(tableName);
+      if (!walog) {
+        env.getClient().tableOperations().setProperty(tableName, Property.TABLE_DURABILITY.getKey(),
+            "none");
+      }
+    } catch (TableExistsException tee) {}
+
+    // scan just to wait for tablet be online
+    Scanner scanner = env.getClient().createScanner(tableName, Authorizations.EMPTY);
+    for (Map.Entry<Key,Value> entry : scanner) {
+      entry.getValue();
+    }
+
+    // number of batches each thread should write
+    int numToWrite = 100;
+
+    ArrayList<WriteTask> wasks = new ArrayList<WriteTask>();
+    ArrayList<Thread> threads = new ArrayList<Thread>();
+
+    BatchWriter bw = null;
+    SharedBatchWriter sbw = null;
+
+    if (useSharedBW) {
+      bw = env.getClient().createBatchWriter(tableName,
+          new BatchWriterConfig().setMaxWriteThreads(1));
+      sbw = new SharedBatchWriter(bw);
+    }
+
+    for (int i = 0; i < numThreads; i++) {
+      WriteTask wask;
+
+      if (useSharedBW) {
+        wask = new WriteTask(sbw, numToWrite, numToBatch);
+      } else {
+        wask = new WriteTask(env.getClient().createBatchWriter(tableName,
+            new BatchWriterConfig().setMaxWriteThreads(1)), numToWrite, numToBatch);
+      }
+
+      wasks.add(wask);
+      Thread thread = new Thread(wask);
+      threads.add(thread);
+    }
+
+    for (Thread thread : threads) {
+      thread.start();
+    }
+
+    for (Thread thread : threads) {
+      thread.join();
+    }
+
+    if (useSharedBW)
+      bw.close();
+
+    long sum = 0;
+    for (WriteTask writeTask : wasks) {
+      sum += writeTask.getTime();
+    }
+
+    int totalNumMutations = numToWrite * numThreads * numToBatch;
+
+    double rate = totalNumMutations / (sum / (double) wasks.size());
+
+    report.info("time", sum / (double) wasks.size(),
+        "Time it took the task to run in milliseconds");
+    report.info("threads", numThreads, "Number of threads");
+    report.info("batch", numToBatch, "Number of batchs");
+    report.info("mutations", totalNumMutations, "Total number of mutations");
+    report.info("rate", rate, "Mutations per millisecond");
+
```

Comment:
It seems like it will be hard to tell these apart in the test output because of the same id.  Could add a one up counter to the ids like :

```java
// testRun is a counter that is passed to method and incremented for every test
report.info("time"+testRun, sum / (double) wasks.size(),
￼         "Time it took the task to run in milliseconds");
￼     report.info("threads"+testRun, numThreads, "Number of threads");
```

It would also be useful to include `numToWrite`, `walog`, and `useSharedBW` in the report

Reply:
I was thinking the same thing actually.  I'll update it.